src/index.ts

```diff
@@ -69,6 +69,13 @@ const getQueryFromFilter = (filter: Filter, customQuery: string): string => {
   }
 }
 
+const isValidCurrentGraph = async (): Promise<boolean> => {
+  const settings = logseq.settings as Settings
+  const currentGraph = await logseq.App.getCurrentGraph()
+
+  return currentGraph?.name === settings.graph
```

Comment:
And settings.graph gets set the first time you sync, right?

Reply:
Oh, is that what is displayed in settings? That is probably ok then.

Reply:
Yeah. Actually, we will trigger an initial sync when the user installs the plugin so basically they are the same 😄

Reply:
Yeah that is what I was thinking, seems to happen right away when reloading.